src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -75,9 +75,9 @@ public boolean add(SelectionKey key)
       }
       keys[pos++] = key;
       if (pos == keys.length) {
-        SelectionKey[] keys = new SelectionKey[this.keys.length << 1];
-        System.arraycopy(this.keys, 0, keys, 0, pos);
-        this.keys = keys;
+        SelectionKey[] lKeys = new SelectionKey[this.keys.length << 1];
```

Comment:
keys is also member variable which gets shadowed. I want to avoid that.

—
Chetan


From: Vlad Rozov <notifications@github.com<mailto:notifications@github.com>>
Reply-To: DataTorrent/Netlet <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, January 13, 2016 at 6:51 AM
To: DataTorrent/Netlet <Netlet@noreply.github.com<mailto:Netlet@noreply.github.com>>
Cc: Chetan Narsude <chetan@apache.org<mailto:chetan@apache.org>>
Subject: Re: [Netlet] make the tests pass by binding to available port as opposed to fixed … (#41)


In src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java<https://github.com/DataTorrent/Netlet/pull/41#discussion_r49598759>:

> @@ -75,9 +75,9 @@ public boolean add(SelectionKey key)
>        }
>        keys[pos++] = key;
>        if (pos == keys.length) {
> -        SelectionKey[] keys = new SelectionKey[this.keys.length << 1];
> -        System.arraycopy(this.keys, 0, keys, 0, pos);
> -        this.keys = keys;
> +        SelectionKey[] lKeys = new SelectionKey[this.keys.length << 1];


Why rename to lKeys is necessary?

—
Reply to this email directly or view it on GitHub<https://github.com/DataTorrent/Netlet/pull/41/files#r49598759>.